### DISCUSSION TITLE
Poetry not installing lxml

### DISCUSSION THREAD

**Discussion by MorganSP** [python-poetry/poetry#8114] [Q&A] [answered]
I have an existing project which uses a vagrant environment but I have problems builing dependencies.nix since pypi2nix has been abandoned. 
So I'm trying to add Poetry to my project but Poetry fails to install lxml 4.7.1. 

`Note: This error originates from the build backend, and is likely not a problem with poetry but with lxml (4.7.1) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "lxml `

It looks like lxml needs libxml2 and libxslt, so i've installed these using Brew, but still the same problem.
The whole output;
  • Installing lxml (4.7.1): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel
  
  Building lxml version 4.7.1.
  Building without Cython.
  Building against libxml2 2.9.4 and libxslt 1.1.29
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-12-x86_64-cpython-311
  creating build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/_elementpath.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/sax.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/pyclasslookup.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/builder.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/doctestcompare.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/usedoctest.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/cssselect.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/ElementInclude.py -> build/lib.macosx-12-x86_64-cpython-311/lxml
  creating build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  creating build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/soupparser.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/defs.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/_setmixin.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/clean.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/_diffcommand.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/html5parser.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/formfill.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/builder.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/ElementSoup.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/_html5builder.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/usedoctest.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  copying src/lxml/html/diff.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/html
  creating build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron
  copying src/lxml/isoschematron/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron
  copying src/lxml/etree.h -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/etree_api.h -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/lxml.etree.h -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/lxml.etree_api.h -> build/lib.macosx-12-x86_64-cpython-311/lxml
  copying src/lxml/includes/xmlerror.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/c14n.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/xmlschema.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/__init__.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/schematron.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/tree.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/uri.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/etreepublic.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/xpath.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/htmlparser.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/xslt.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/config.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/xmlparser.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/xinclude.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/dtdvalid.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/relaxng.pxd -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/lxml-version.h -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  copying src/lxml/includes/etree_defs.h -> build/lib.macosx-12-x86_64-cpython-311/lxml/includes
  creating build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources
  creating build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/rng
  copying src/lxml/isoschematron/resources/rng/iso-schematron.rng -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/rng
  creating build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl
  copying src/lxml/isoschematron/resources/xsl/XSD2Schtrn.xsl -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl
  copying src/lxml/isoschematron/resources/xsl/RNG2Schtrn.xsl -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl
  creating build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
  copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/iso_abstract_expand.xsl -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
  copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/iso_dsdl_include.xsl -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
  copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/iso_schematron_skeleton_for_xslt1.xsl -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
  copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/iso_svrl_for_xslt1.xsl -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
  copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/iso_schematron_message.xsl -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
  copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/readme.txt -> build/lib.macosx-12-x86_64-cpython-311/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
  running build_ext
  building 'lxml.etree' extension
  creating build/temp.macosx-12-x86_64-cpython-311
  creating build/temp.macosx-12-x86_64-cpython-311/src
  creating build/temp.macosx-12-x86_64-cpython-311/src/lxml
  clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -DCYTHON_CLINE_IN_TRACEBACK=0 -Isrc -Isrc/lxml/includes -I/private/var/folders/qf/z97_2t7138v_mh726mmdhq980000gn/T/tmplckbses6/.venv/include -I/usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c src/lxml/etree.c -o build/temp.macosx-12-x86_64-cpython-311/src/lxml/etree.o -w -flat_namespace
  src/lxml/etree.c:262403:23: error: no member named 'exc_type' in 'struct _err_stackitem'
      while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
              ~~~~~~~~  ^
  src/lxml/etree.c:262403:53: error: no member named 'exc_type' in 'struct _err_stackitem'
      while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
                                            ~~~~~~~~  ^
  src/lxml/etree.c:262417:23: error: no member named 'exc_type' in 'struct _err_stackitem'
      *type = exc_info->exc_type;
              ~~~~~~~~  ^
  src/lxml/etree.c:262419:21: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      *tb = exc_info->exc_traceback;
            ~~~~~~~~  ^
  src/lxml/etree.c:262433:26: error: no member named 'exc_type' in 'struct _err_stackitem'
      tmp_type = exc_info->exc_type;
                 ~~~~~~~~  ^
  src/lxml/etree.c:262435:24: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      tmp_tb = exc_info->exc_traceback;
               ~~~~~~~~  ^
  src/lxml/etree.c:262436:15: error: no member named 'exc_type' in 'struct _err_stackitem'
      exc_info->exc_type = type;
      ~~~~~~~~  ^
  src/lxml/etree.c:262438:15: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      exc_info->exc_traceback = tb;
      ~~~~~~~~  ^
  src/lxml/etree.c:262520:30: error: no member named 'exc_type' in 'struct _err_stackitem'
          tmp_type = exc_info->exc_type;
                     ~~~~~~~~  ^
  src/lxml/etree.c:262522:28: error: no member named 'exc_traceback' in 'struct _err_stackitem'
          tmp_tb = exc_info->exc_traceback;
                   ~~~~~~~~  ^
  src/lxml/etree.c:262523:19: error: no member named 'exc_type' in 'struct _err_stackitem'
          exc_info->exc_type = local_type;
          ~~~~~~~~  ^
  src/lxml/etree.c:262525:19: error: no member named 'exc_traceback' in 'struct _err_stackitem'
          exc_info->exc_traceback = local_tb;
          ~~~~~~~~  ^
  src/lxml/etree.c:262711:26: error: no member named 'exc_type' in 'struct _err_stackitem'
      tmp_type = exc_info->exc_type;
                 ~~~~~~~~  ^
  src/lxml/etree.c:262713:24: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      tmp_tb = exc_info->exc_traceback;
               ~~~~~~~~  ^
  src/lxml/etree.c:262714:15: error: no member named 'exc_type' in 'struct _err_stackitem'
      exc_info->exc_type = *type;
      ~~~~~~~~  ^
  src/lxml/etree.c:262716:15: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      exc_info->exc_traceback = *tb;
      ~~~~~~~~  ^
  src/lxml/etree.c:264915:20: error: no member named 'exc_type' in 'struct _err_stackitem'
      t = exc_state->exc_type;
          ~~~~~~~~~  ^
  src/lxml/etree.c:264917:21: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      tb = exc_state->exc_traceback;
           ~~~~~~~~~  ^
  src/lxml/etree.c:264918:16: error: no member named 'exc_type' in 'struct _err_stackitem'
      exc_state->exc_type = NULL;
      ~~~~~~~~~  ^
  fatal error: too many errors emitted, stopping now [-ferror-limit=]
  20 errors generated.
  Compile failed: command '/usr/bin/clang' failed with exit code 1
  creating var
  creating var/folders
  creating var/folders/qf
  creating var/folders/qf/z97_2t7138v_mh726mmdhq980000gn
  creating var/folders/qf/z97_2t7138v_mh726mmdhq980000gn/T
  cc -I/usr/include/libxml2 -c /var/folders/qf/z97_2t7138v_mh726mmdhq980000gn/T/xmlXPathInitqbdzenlw.c -o var/folders/qf/z97_2t7138v_mh726mmdhq980000gn/T/xmlXPathInitqbdzenlw.o
  cc var/folders/qf/z97_2t7138v_mh726mmdhq980000gn/T/xmlXPathInitqbdzenlw.o -lxml2 -o a.out
  error: command '/usr/bin/clang' failed with exit code 1
  

  at ~/Library/Application Support/pypoetry/venv/lib/python3.11/site-packages/poetry/installation/chef.py:152 in _prepare
      148│ 
      149│                 error = ChefBuildError("\n\n".join(message_parts))
      150│ 
      151│             if error is not None:
    → 152│                 raise error from None
      153│ 
      154│             return path
      155│ 
      156│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

I should add that I am on Mac OS

**Comment by clintonroy** [comment]
This is very clearly a C compilation problem, nothing to do with poetry. You'll (almost certainly) get the same issue with pip.

**Reply by MorganSP** [reply]
Thanks. I'm just wondering how/where I can find a solution.

**Reply by clintonroy** [reply, marked as answer]
Well, I'd try installing the latest lxml if you can.

**Reply by MorganSP** [reply]
Thanks. I actually already tried this. I got a similar error but on closer inspection it was regarding another package. My mistake.
That worked.

**Reply by nbro10** [reply]
I'm getting this problem while doing `poetry add mypy@latest --group=dev`, which attempts to install mypy 1.5.1. I don't have `lxml` specified in my dependencies, so it's a dependency of a dependency, in my case `zeep` (>=4.1.0,<5.0.0), which a dependency of another dep. MacOS ventura.
